### DISCUSSION TITLE
Lesson 6: network.show_active() is None when running brownie test

### DISCUSSION THREAD

**Discussion by dbrans** [smartcontractkit/full-blockchain-solidity-course-py#719] [Q&A] [answered]
Hi,
When I run `brownie run deploy.py --network ganache-local` everything works fine. 
However, when I run `brownie test` my `network.show_active()` is `None`, even if I manually try to set the network to something like ganache-local.

Any ideas?
Thank you in advance!

**Comment by cromewar** [comment]
Hello @dbrans could you please share your test script and the command you are using to run it?

**Comment by albert-garreta** [comment]
I am having the exact same issue. I can share my own scripts if necessary.

**Reply by cromewar** [reply]
Yes please, share your helpful scripts and your brownie config.yaml

**Reply by albert-garreta** [reply]
Hi,

This is the brownie-config.yaml
<img width="1280" alt="brownie-config 1" src="https://user-images.githubusercontent.com/97292142/148690786-da3b536f-d00a-4c33-8405-e851dec896e2.png">

Here is the test script
<img width="1280" alt="test" src="https://user-images.githubusercontent.com/97292142/148690799-e2ba3b71-15ee-47d0-8a3f-e05db2fcee40.png">

And here is the error I get when doing "brownie test"
<img width="1280" alt="brownie-config 2" src="https://user-images.githubusercontent.com/97292142/148690821-5c3a142f-eecd-4686-a979-e3e9103ad3b7.png">

Thanks in advance!

**Reply by cromewar** [reply]
Yep I think the error is the follwing: 

The line: 

```Python
price_feed_address = config["networks"][network.show_active()][
            "eth_usd_price_feed"
        ]
```

But  `eth_usd_price_feed` is not on your brownie_config on the development network, so add it and it should work

**Reply by LintonAchmad** [reply]
Correct, your `default` network which points to `development` doesn't have the `eth_usd_price_feed` - add that and it should work.

**Reply by albert-garreta** [reply]
Hi,  thanks for your replies :)

That was not the problem. If using a local development network the code deploys a Mock price feed and sets `price_feed_address` to be the Mock's address. (I had the same problem when testing on rinkeby for example, where `eth_usd_price_feed` is defined in the brownie-config)

I fond out what is going on. It turns out that `network.show_active()` returns `None` until a network is launched or connected to. In test mode, this only occurs when the test executes the first test function. In my case the imports were done before the test functions are executed. At that point `network.show_active()` was `None`, giving rise to an error.

To fix this I moved the problematic imports and the `get_account()` call inside the test functions. Here is the fixed test script:

<img width="1280" alt="fix" src="https://user-images.githubusercontent.com/97292142/148759779-0079f8f2-c3c6-425c-80d3-24fa6a4d91bb.png">

**Comment by cromewar** [comment, marked as answer]
@dbrans  Good job! this could be helpful for others too.

Closing this thread as is now solved.
